http/cors.go

```diff
@@ -0,0 +1,32 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/vault"
+)
+
+func wrapCORSHandler(h http.Handler, core *vault.Core) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		corsConf := core.CORSConfig()
```

Comment:
This is in the hot path since it's called on every request, so we should first check whether cors is even enabled, and immediately chain to the next handler if not. We don't really need to acquire a lock here because if cors is being toggled and requests are flying in, we don't have guaranteed ordering/timing anyways.

If cors is enabled, do we actually need to send those headers back on *every* request, or just with OPTIONS?